### PR TITLE
Alter authorization on database to roles should be blocked

### PR DESCRIPTION
**Pull request by shalinilohia50**
### Description

`Alter authorization on database` to any database role/fixed server role/sysadmin should be blocked. Earlier, it was checking if the role exists in the pg_authid for login. This was returning true for database role/fixed server role/sysadmin which is not the expected TSQL behaviour.

### Issues Resolved

JIRA: BABEL-6122
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

** BABEL-SP_TABLE_PRIVILIGES-vu-verify FAILED! Execution timed out!!

### CODE REVIEW

**Review by HarshLunagariya — APPROVED** [review]

File: test/JDBC/expected/alter_authorization_change_db_owner-vu-verify.out [hunk 1]
```diff
@@ -1185,6 +1185,40 @@ use master
...
+
+~~ERROR (Message: An entity of type database cannot be owned by a role, a group, an approle, or by principals mapped to certificates or asymmetric keys.)~~
+
+alter authorization on database::alter_auth_db TO dbcreator
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: An entity of type database cannot be owned by a role, a group, an approle, or by principals mapped to certificates or asymmetric keys.)~~
+
+alter authorization on database::alter_auth_db TO master_db_datareader
```
Comment (anju15bharti) [inline test/JDBC/expected/alter_authorization_change_db_owner-vu-verify.out:1247]: 1. Add test cases for user-defined roles, ensuring they fail as expected. Additionally, include test cases for all system-defined database roles to comprehensively cover negative scenarios.
2. Add equivalent test cases for sp_changedbowner, covering the same conditions

**Review by anju15bharti — CHANGES_REQUESTED** [review]

File: test/JDBC/input/alter_authorization_change_db_owner-vu-verify.mix [hunk 2]
```diff
@@ -681,10 +681,41 @@ go
...
+go
+alter authorization on database::alter_auth_db TO db_ddladmin
+go
+alter authorization on database::alter_auth_db TO dbo
+go
+alter authorization on database::alter_auth_db TO db_owner
+go
+alter authorization on database::alter_auth_db TO master_db_datareader
+go
+alter authorization on database::alter_auth_db TO alter_auth_r1
```
Comment (anju15bharti) [inline test/JDBC/input/alter_authorization_change_db_owner-vu-verify.mix:713]: We should add test cases prepending database name as it was falsely working before this fix.

**Review by anju15bharti — CHANGES_REQUESTED** [review]

**Review by anju15bharti — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]
